dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -419,6 +416,25 @@ public void taskFinished(TaskInstance taskInstance) throws StateEventHandleExcep
         }
     }
 
+    /**
+     * continue to submit subsequent tasks after failure
+     *
+     * if there are subsequent tasks:
+     *      1. nodes are parallel;
+     *      2. nodes are serial and failed task type is condition;
+     *      3. nodes are serial and next task type is condition;
+     */
+    private void failureContinueSubmitPostNodes(TaskInstance taskInstance) throws StateEventHandleException {
+        if (!DagHelper.haveAnyNodeAfterNode(Long.toString(taskInstance.getTaskCode()), dag)) {
+            submitPostNode(Long.toString(taskInstance.getTaskCode()));
+        } else {
+            if (taskInstance.isConditionsTask()
+                    || DagHelper.haveConditionsAfterNode(Long.toString(taskInstance.getTaskCode()), dag)) {
+                submitPostNode(Long.toString(taskInstance.getTaskCode()));
+            }
+        }
```

Comment:
could we move these conditions to a private function or workflow util, such as `shouldSubmitPostNode`? because all these condition should submitPostNode, might be clearer for reading. WDYT.

Reply:
That's excellent, i will improve it.